python/src/array.cpp

```diff
@@ -624,6 +624,18 @@ void init_array(py::module_& m) {
             return divide(to_array(v, float32), a);
           },
           "other"_a)
+      .def(
+          "__mod__",
+          [](const array& a, const ScalarOrArray v) {
+            return remainder(a, to_array(v, a.dtype()));
+          },
+          "other"_a)
```

Comment:
`b` is made into a `float32` array and then in the remainder op they are both cast to `promote_dtypes(int32, float32)` so it works as expected and we get a float32. Good thinking though cause I just assumed it would work and didn't check.

Reply:
Right, I forgot that's what `to_array` does. Thanks!